var/spack/repos/builtin/packages/py-aniso8601/package.py

```diff
@@ -13,5 +13,8 @@ class PyAniso8601(PythonPackage):
     pypi = "aniso8601/aniso8601-9.0.1.tar.gz"
 
     version("9.0.1", sha256="72e3117667eedf66951bb2d93f4296a56b94b078a8a95905a052611fb3f1b973")
+    version("7.0.0", sha256="513d2b6637b7853806ae79ffaca6f3e8754bdd547048f5ccc1420aec4b714f1e")
 
+    depends_on("python@3.7:3.9", type=("build", "run"), when="@9.0.1")
+    depends_on("python@3.7", type=("build", "run"), when="@7.0.0")
```

Comment:
Why do you think newer versions of Python won't work? From testing?

Reply:
I don't know if newer version of Python will work.
I just read the setup.py:
* line 50 at https://bitbucket.org/nielsenb/aniso8601/src/v9.0.1/setup.py
* line 40 at https://bitbucket.org/nielsenb/aniso8601/src/v7.0.0/setup.py
Isn't that enough? Do I need to test each version of python for each python package recipe I want to add? If so, I will need functional tests because the version incompatibilities are not all visible when the python module is loaded.

Reply:
Those comments don't mean anything, it's basically just which versions of Python the developer is testing. They don't prevent pip from installing with newer Python, and Python is usually backwards compatible enough that it will still work for newer versions. I wouldn't restrict things unless you know for a fact that they won't work.

Reply:
Thank you for the information and help provided. It actually seems more reasonable to me not to over-constrain the recipe. I removed the python dependency (after testing the installation in python 3.8.0, 3.9.0, 3.10.0 and 3.11.0 with the `spack test` command)